src/otx/v2/api/core/dataset.py

```diff
@@ -315,8 +315,10 @@ def set_datumaro_adapters(self, data_roots: str | None = None) -> None:
             unlabeled_data_roots=self.unlabeled_data_roots,
             unlabeled_file_list=self.unlabeled_file_list,
         )
-        self.label_schema: LabelSchemaEntity = self.dataset_adapter.get_label_schema()
         self.dataset_entity: dict[Subset, DatumDataset] = self.dataset_adapter.get_otx_dataset()
+        # NOTE: sequence matters as label_entities is changed in get_otx_dataset()
+        # I'd suggest not to tamper with labels in get_otx_dataset() as unexpected behavior may occur
+        self.label_schema: LabelSchemaEntity = self.dataset_adapter.get_label_schema()
```

Comment:
Order should be get_label_schema() -> get_otx_dataset()

When get_lable_schema() -> get_otx_dataset() scenario, there is a bug that OTX will show unused categories in validation metric. 
![image](https://github.com/openvinotoolkit/training_extensions/assets/106788349/10546c55-b5b8-4236-9509-72cef74402fd)

When get_otx_dataset() -> get_label_schema() scenario, there is a bug that OTX(Datumaro adapters) will delete used categories.
![image](https://github.com/openvinotoolkit/training_extensions/assets/106788349/efb9f481-4410-48df-9593-c3ba6994bf9f)
I think the second bug looks more severe, so let's keep the procedure as it is and fix this bug in later phase

Reply:
@eugene123tw, one more vote for @jaegukhyun's opinion. Please remove `get_otx_dataset()` -> `get_label_schema()` scenario.

Reply:
DONE